tests/ci.rs

```diff
@@ -40,18 +41,25 @@ fn generate() {
                 .add_type(PullRequestType::Opened)
                 .add_type(PullRequestType::Synchronize)
                 .add_type(PullRequestType::Reopened)
+                .add_type(PullRequestType::Closed)
                 .add_branch("main"),
         );
 
-    let permissions = Permissions::default()
-        .pull_requests(Level::Write)
-        .packages(Level::Write)
-        .contents(Level::Write);
+    let deploy = Job::new("Deploy to Shuttle")
+        .permissions(Permissions::default().contents(Level::Write))
+        .cond(Expression::new("github.event.pull_request.merged == true"))
```

Comment:
use Context API for this and make it run on push to main Instead of merged pull request

Reply:
https://github.com/tailcallhq/gh-workflow/blob/11e5e82bbb2038b2bd1da67ee19a62d6585e8c7a/tests/ci.rs#L55